lib/Yield/Assembly/DisassemblerInstance.cpp

```diff
@@ -0,0 +1,338 @@
+  Context = std::make_unique<llvm::MCContext>(AssemblyInformation.get(),
+                                              RegisterInformation.get(),
+                                              ObjectFileInformation.get());
+
+  llvm::Triple Triple(Architecture);
+  bool IsPIC = false;
+  ObjectFileInformation->InitMCObjectFileInfo(Triple, IsPIC, *Context);
+
+  auto &SI = *SubtargetInformation;
+  Disassembler.reset(LLVMTarget->createMCDisassembler(SI, *Context));
+  revng_assert(Disassembler != nullptr, "Disassembler object creation failed.");
+
+  InstructionInformation.reset(LLVMTarget->createMCInstrInfo());
+
+  unsigned AssemblyDialect = 0;
+  if (*LLVMArchitecture == llvm::Triple::ArchType::x86
+      || *LLVMArchitecture == llvm::Triple::ArchType::x86_64)
+    if (options::UseIntelSyntax == true)
+      AssemblyDialect = 1;
+  Printer.reset(LLVMTarget->createMCInstPrinter(Triple,
+                                                AssemblyDialect,
+                                                *AssemblyInformation,
+                                                *InstructionInformation,
+                                                *RegisterInformation));
+  revng_assert(Printer != nullptr, "Printer object creation failed.");
+
+  using namespace options;
+  if (ImmediateStyle == ImmediateStyles::Decimal) {
+    Printer->setPrintImmHex(false);
+    Printer->setPrintBranchImmAsAddress(false);
+  } else {
+    Printer->setPrintImmHex(true);
+    Printer->setPrintBranchImmAsAddress(true);
+  }
+
+  if (ImmediateStyle == ImmediateStyles::CHexadecimal)
+    Printer->setPrintHexStyle(llvm::HexStyle::C);
+  else if (ImmediateStyle == ImmediateStyles::AsmHexadecimal)
+    Printer->setPrintHexStyle(llvm::HexStyle::Asm);
+
+  Printer->setSymbolizeOperands(ShouldSymbolizeOperands);
+  Printer->setUseMarkup(true);
+}
+
+static std::pair<std::optional<llvm::MCInst>, uint64_t>
+getInstruction(const MetaAddress &Address,
+               llvm::ArrayRef<uint8_t> RawBytes,
+               const llvm::MCDisassembler &Disassembler) {
+  llvm::MCInst Result;
+  llvm::raw_null_ostream NullStream;
+  uint64_t LocalSize = 0;
+  bool Success = Disassembler.getInstruction(Result,
+                                             LocalSize,
+                                             RawBytes,
+                                             Address.asPC(),
+                                             NullStream);
+  if (Success == false || LocalSize == 0)
+    return { std::nullopt, 0 };
+
+  return { std::move(Result), LocalSize };
+}
+
+// static std::optional<size_t>
+// findLastOpenTag(const assembly::Instruction::Fragment &Fragment) {
+//   for (size_t Index = Fragment.Tags.size() - 1;
+//        Index != std::numeric_limits<size_t>::max();
+//        --Index)
+//     if (Fragment.Tags[Index].ToIndex == std::numeric_limits<size_t>::max())
+//       return Index;
+//
+//   return std::nullopt;
+// }
```

Comment:
Why the commented code? We don't keep commented code.

Reply:
Whoops, this is obsolete. I just forgot to remove it.

Reply:
There's more commented code below, take that out too.

Reply:
Yeah, sorry.